internal/provider/graphite/provider.go

```diff
@@ -59,6 +61,8 @@ func (provider *provider) Fetch(ctx context.Context, prefix string, last time.Du
 	q.Add(queryParam, prefix)
 	request.URL.RawQuery = q.Encode()
 
+	provider.listener.OnStepQueued()
```

Comment:
I'm not sure that it's correct. Why not in fetch/recursive?

Reply:
I moved it into fetch